doc/development/architecture.rst

```diff
@@ -0,0 +1,315 @@
+It provides a flexible and modular architecture that allows users to easily extend and customize the simulation setup.
+
+The architecture provides following features to achieve the goals outlined in the :doc:`goals and features <../goals_features>` document:
+
+- Easy coupling of multiple domains and physics
+- Field and model initialization based on dependency graphs
+- Modular solver design that computes the data dependencies at runtime
+- Plugin architecture for easy extension with new models and fields
+
+The main abstraction is that each domain has one physics module assigned to it that defines the governing equations and models for that domain.
+Multiple domains can be defined with input files and the coupling between the domains is handled automatically based on the defined physics modules.
+
+The following sections describe the main feature and and implementation example to give a high level overview of the architecture.
+More details are described in the section for each feature. 
+
+Solver Execution Model
+----------------------
+
+FoamAdapter defines solver behavior as a dependency-driven execution model.
+Each solver declares its required fields and models, and the runtime system constructs a directed acyclic graph (DAG) to determine the correct execution order.
+
+The following diagram illustrates the workflow for two common solver types for a conjugate heat transfer scenario:
+
+.. mermaid::
+
+    %%{init: {'flowchart': { 'htmlLabels': true, 'wrap': true }}}%%
+    flowchart TB
+        subgraph setup1 ["setup "]
+            direction TB
+            S1A["Initialize Fields"]
+        end
+
+        subgraph setup2 ["setup"]
+            direction TB
+            S2A["Initialize Temperature Field"]
+        end
+
+        subgraph S1 ["non-thermal-fluid solver"]
+            direction TB
+            S1C["Momentum Predictor"]
+            S1C --> S1E["Solve Energy Equation"]
+            S1E --> S1F["Pressure Corrector<br/>PISO Loop"]
+            S1F --> S1J["Update turbulence Model"]
+        end
+
+        subgraph S2 ["thermal-solid solver"]
+            direction TB
+            S2B["Solve Energy Equation"] --> S2D["Update Solid Properties"]
+        end
+
+        setup1 --> S1C
+        setup2 --> S2B
+        S1E --> S2B
+        S2B --> S1E
+
+
+The diagram shows two selected solvers: a non-thermal-fluid solver and a thermal-solid solver.
+Each solver has its own setup phase and computational steps, with interactions between the two solvers for energy exchange.
+
+The solution steps of each solver provide the dependencies and the runtime system constructs a DAG to determine the correct execution order based on these dependencies.
+
+.. code-block:: python
+
+    # 'model' is a placeholder for the model dependency object required by each step.
+    model = ...  # placeholder for model dependency
+    @model("fluidSolver")
+    class DomainA:
+        @step(order=1, model)
```

Comment:
Does it really need `model` here? Maybe to be a real DAG `depends_on`  would be required.